balloon/tests/balloon_m.rs

```diff
@@ -0,0 +1,102 @@
+#![cfg(feature = "alloc")]
```

Comment:
I think you can remove the `alloc` requirement by using `hash_with_memory` and allocating the necessary buffer inside of the test function. Test modules always have access to `std`.